persist/adapter.go

```diff
@@ -56,6 +56,9 @@ type Adapter interface {
 	// RemovePolicy removes a policy rule from the storage.
 	// This is part of the Auto-Save feature.
 	RemovePolicy(sec string, ptype string, rule []string) error
+	// UpdatePolicy updates a policy rule from storage.
+	// This is part of the Auto-Save feature.
+	UpdatePolicy(sec string, ptype string, oldRule, newPolicy []string) error
```

Comment:
Don't add new funcs here. Define a new interface called UpdatableAdapter, add this function to it.

Reply:
@hsluoyz What do you think of merging this PR into beta branch.

Reply:
@nodece that sounds OK to me.

Reply:
@hsluoyz We made a lot of patches, like BatchAdapter. Do you have any good ideas? Maybe we'll make a lot of breaking changes.

Reply:
What're the patches for **BatchAdapter**?

Reply:
@hsluoyz https://github.com/casbin/casbin/blob/master/internal_api.go#L143

Reply:
The patches look ugly but it's OK for me. It works well to support all kinds of adapters and doesn't bring burden or breaking changes to users.

Reply:
> @00LT00 Don't add new funcs here. Define a new interface called UpdatableAdapter, add this function to it.

@00LT00 please add `UpdatableAdapter` to Casbin codebase.

Reply:
ok